### PR TITLE
MMA-Date picker proposal

### PR DESCRIPTION
**Pull request by AlbertoMApps**
First proposal of date picker based on:
https://github.com/PranavMaganti/compose-material-dialogs?tab=readme-ov-file
https://www.youtube.com/watch?v=BhyavkT2UO4

### CODE REVIEW

**Comment by jchodev** [comment]
if we are only using for date picker, we can implement origin material3 datepicker instead of library
- https://www.composables.com/material3/datepicker
- https://medium.com/@rahulchaurasia3592/material3-datepicker-and-datepickerdialog-in-compose-in-android-54ec28be42c3

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    DatePickerDialog(
+        onDismissRequest = { onDismiss() },
+        confirmButton = {
+            Button(onClick = {
+                onDateSelected(selectedDate)
+                onDismiss()
+            }
+
+            ) {
+                Text(text = "OK")
```
Comment (jchodev) [inline core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt]: maybe pass from outside? and default with android.R.string.ok

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        },
+        dismissButton = {
+            Button(onClick = {
+                onDismiss()
+            }) {
+                Text(text = "Cancel")
```
Comment (jchodev) [inline core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt]: maybe pass from outside? and default with android.R.string.cancel

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    ) {
+        DatePicker(
+            state = datePickerState
+        )
+    }
+}
+
+private fun convertMillisToDate(millis: Long): String {
+    val formatter = SimpleDateFormat("dd/MM/yyyy")
```
Comment (jchodev) [inline core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt]: will we allow user to change the date format ?

Comment (AlbertoMApps) [reply core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt]: I don't think it's needed for now.

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt (hunk 1 above)
Comment (AlbertoMApps) [reply core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt]: That can be an option, but it really depends if that string changes or not.
I think using R.string.ok or cancel is better.

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt (hunk 3 above)
Comment (jchodev) [reply core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/DatePickerPromptDialog.kt]: I think constant with this, that we can reuse at whole application

File: feature/calendar/src/main/java/com/jerryalberto/mmas/feature/calendar/ui/screen/CalendarScreen.kt [hunk 4]
```diff
@@ -1,32 +1,67 @@
...
+    var date by remember {
+        mutableStateOf("dd/MM/yyyy")
+    }
+
+    var showDatePicker by remember {
+        mutableStateOf(false)
+    }
+
+    Button(onClick = { showDatePicker = true }) {
+        Text(text = "Open date picker dialog", textAlign = TextAlign.Center)
```
Comment (jchodev) [inline feature/calendar/src/main/java/com/jerryalberto/mmas/feature/calendar/ui/screen/CalendarScreen.kt]: string.xml with "Open date picker dialog"

**Review by jchodev — APPROVED** [review]

**Pull request merged**